### PR TITLE
HDDS-2766. security/SecuringDataNodes.md

### PR DESCRIPTION
**Pull request by iamabug**
## What changes were proposed in this pull request?

translation to doc https://hadoop.apache.org/ozone/docs/0.5.0-beta/security/securingdatanodes.html

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-2766

## How was this patch tested?

hugo server

### CODE REVIEW

File: hadoop-hdds/docs/content/security/SecuringDatanodes.zh.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+---
+title: "安全化 Datanode"
+date: "2019-April-03"
+weight: 2
+summary:  解释安全化 datanode 的不同模式，包括 Kerberos、证书的手动颁发和自动颁发等。
```
Comment (iamabug) [inline hadoop-hdds/docs/content/security/SecuringDatanodes.zh.md:5]: Question about original doc, which is more precise, `datanode` or `data node` ?

File: hadoop-hdds/docs/content/security/SecuringDatanodes.zh.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+## 如何安全化 datanode
+
+在 Ozone 中，当 datanode 启动并发现 SCM 的地址之后，datanode 要做的第一件事就是创建私钥并向 SCM 发送证书请求。
+
+<h3>通过 Kerberos 颁发证书<span class="badge badge-secondary">当前模型</span></h3>
+SCM 有一个内置的 CA 用来批准证书请求，如果 datanode 已经有一个 Kerberos keytab，SCM 会信任它并自动颁发一个证书。
+
+
+<h3>手动颁发<span class="badge badge-primary">开发中</span></h3>
+如果 datanode 是新加入的并且没有 keytab，那么证书请求需要等待管理员的批准。换句话说，信任关系链由集群管理员建立。
```
Comment (iamabug) [inline hadoop-hdds/docs/content/security/SecuringDatanodes.zh.md:47]: Original doc says `the web of trust`, but according to my understanding, it is called chain of trust in PKI.

Comment (iamabug) [reply hadoop-hdds/docs/content/security/SecuringDatanodes.zh.md:47]: A typo in origin doc maybe, 'band new` should be `brand new` ? If yes, should I fix it in this PR or open a new one ?

**Comment by iamabug** [comment]
@xiaoyuyao @cxorm Could you help review this ? Thanks. BTW, I cannot find request review button on this page.

File: hadoop-hdds/docs/content/security/SecuringDatanodes.zh.md (hunk 1 above)
Comment (xiaoyuyao) [reply hadoop-hdds/docs/content/security/SecuringDatanodes.zh.md:5]: datanode

File: hadoop-hdds/docs/content/security/SecuringDatanodes.zh.md (hunk 2 above)
Comment (xiaoyuyao) [reply hadoop-hdds/docs/content/security/SecuringDatanodes.zh.md:47]: this should be "brand new" you can fix it in this one along with the CN doc.

Comment (xiaoyuyao) [reply hadoop-hdds/docs/content/security/SecuringDatanodes.zh.md:47]: "chain of trust" is better, pls fix it in both EN and CN doc.

**Comment by xiaoyuyao** [comment]
> @xiaoyuyao @cxorm Could you help review this ? Thanks. BTW, I cannot find request review button on this page.

You can do that on the top right, "reviewers" button. The dropdown list allows you to select reviewers.

File: hadoop-hdds/docs/content/security/SecuringDatanodes.zh.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+
+
+过去，Hadoop 中 datanode 的安全机制是通过在节点上创建 Keytab 文件实现的，而 Ozone 改用 datanode 证书，在安全的 Ozone 集群中，datanode 不再需要 Kerberos。
```
Comment (xiaoyuyao) [inline hadoop-hdds/docs/content/security/SecuringDatanodes.zh.md:26]: Can we rephrase: 
"过去，Hadoop 中 datanode 的安全机制是通过在节点上创建 Keytab 文件实现的，而 Ozone 改用 datanode 证书，在安全的 Ozone 集群中，datanode 不再需要 Kerberos。"
=>
"Hadoop 中 datanode 的安全机制是通过给每个节点创建 Keytab 文件实现的。  Ozone 的datanode安全机制不依赖Kerberos， 而是改用 datanode 证书。

File: hadoop-hdds/docs/content/security/SecuringDatanodes.zh.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+但是我们也支持传统的基于 Kerberos 的认证来方便现有用户，用户只需要在 hdfs-site.xml 里配置下面参数即可：
+
+参数名|描述
+--------|--------------
+dfs.datanode.kerberos.principal| datanode 的服务主体名 <br/> 比如：dn/_HOST@REALM.COM
+dfs.datanode.keytab.file| datanode 进程所使用的 keytab 文件
+hdds.datanode.http.kerberos.principal| datanode http 服务器的服务主体名
```
Comment (xiaoyuyao) [inline hadoop-hdds/docs/content/security/SecuringDatanodes.zh.md:34]: hdds.datanode.http.kerberos.principal=>hdds.datanode.http.auth.kerberos.principal

File: hadoop-hdds/docs/content/security/SecuringDatanodes.zh.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+hdds.datanode.http.kerberos.keytab| datanode http 服务器所使用的 keytab 文件
```
Comment (xiaoyuyao) [inline hadoop-hdds/docs/content/security/SecuringDatanodes.zh.md:35]: hdds.datanode.http.kerberos.keytab=>hdds.datanode.http.auth.kerberos.keytab

Comment (xiaoyuyao) [reply hadoop-hdds/docs/content/security/SecuringDatanodes.zh.md:35]: datanode http 服务器所使用的 keytab 文件=>datanode http 服务器的服务主体登陆所使用的 keytab 文件

File: hadoop-hdds/docs/content/security/SecuringDatanodes.zh.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+
+## 如何安全化 datanode
+
+在 Ozone 中，当 datanode 启动并发现 SCM 的地址之后，datanode 要做的第一件事就是创建私钥并向 SCM 发送证书请求。
```
Comment (xiaoyuyao) [inline hadoop-hdds/docs/content/security/SecuringDatanodes.zh.md:40]: datanode 要做的第一件事就是创建私钥并向 SCM 发送证书请求-》datanode 首先创建私钥并向 SCM 发送证书请求。

File: hadoop-hdds/docs/content/security/SecuringDatanodes.zh.md (hunk 2 above)
Comment (xiaoyuyao) [reply hadoop-hdds/docs/content/security/SecuringDatanodes.zh.md:47]: 那么证书请求需要等待管理员的批准（手动批准功能尚未完全支持）

File: hadoop-hdds/docs/content/security/SecuringDatanodes.zh.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+
+
+<h3>手动颁发<span class="badge badge-primary">开发中</span></h3>
+如果 datanode 是新加入的并且没有 keytab，那么证书请求需要等待管理员的批准。换句话说，信任关系链由集群管理员建立。
+
+<h3>自动颁发 <span class="badge badge-secondary">开发中</span></h3>
+如果你通过 Kubernetes 这样的容器编排软件运行 Ozone，Kubernetes 需要为 datanode 创建一次性 token，用于在启动阶段证明 datanode 容器的身份。（这个特性也正在开发中。）
+
+
+证书颁发后，datanode 的安全就得到了保障，并且 OM 可以颁发块 token。如果 datanode 没有证书或者 SCM 的根证书，datanode 会自动进行注册，下载 SCM 的根证书，并获取自己的证书。
```
Comment (xiaoyuyao) [inline hadoop-hdds/docs/content/security/SecuringDatanodes.zh.md:53]: The original document has a typo here "down load" should be "download"

File: hadoop-hdds/docs/content/security/SecuringDatanodes.zh.md (hunk 4 above)
Comment (iamabug) [reply hadoop-hdds/docs/content/security/SecuringDatanodes.zh.md:34]: Is this a committed change ? I notice that it is still hdds.datanode.http.kerberos.principal in master branch.

**Comment by iamabug** [comment]
> > @xiaoyuyao @cxorm Could you help review this ? Thanks. BTW, I cannot find request review button on this page.
> 
> You can do that on the top right, "reviewers" button. The dropdown list allows you to select reviewers.

I used to have that, but now it looks like this. `Reviewers` is just plain text, I don't think there is a place to click.

File: hadoop-hdds/docs/content/security/SecuringDatanodes.zh.md (hunk 4 above)
Comment (smengcl) [reply hadoop-hdds/docs/content/security/SecuringDatanodes.zh.md:34]: @iamabug Yes. It is in master: https://github.com/apache/hadoop-ozone/blob/471038fe80f44403210a13d352940485ef921c55/hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/HddsConfigKeys.java#L249-L251

Comment (iamabug) [reply hadoop-hdds/docs/content/security/SecuringDatanodes.zh.md]: Thanks for the link and commit. @smengcl

**Comment by cxorm** [comment]
Thanks @iamabug for the work, 
Overall looks great to me (including the fixes), 
I would committed it If all CI checks passed.

**Comment by cxorm** [comment]
Thanks @iamabug for the contribution.
Thanks @xiaoyuyao and @smengcl for reviewing this PR, 
and thanks @adoroszlai for adding the label.

**Comment by ChenSammi** [comment]
Hi @iamabug ,  Can I get your email address? you can sent it to sammichen@apache.org.

**Comment by iamabug** [comment]
> Hi @iamabug , Can I get your email address? you can sent it to [sammichen@apache.org](mailto:sammichen@apache.org).

Hi @ChenSammi , my email is xiangzhang1128@gmail.com

**Pull request merged**